docs/usage.md

```diff
@@ -138,5 +279,14 @@ If you want to load in a new value from your Secret Manager without making any n
 
 You can also use the `argocd app diff` command passing the `--hard-refresh` flag. This will run argocd-vault-plugin again and pull in the new values from you Secret Manager and then you can either have Auto Sync setup or Sync manually to apply the new values.
 
-### Caching the Vault Token
+### Caveats
+
+#### Caching the Hashicorp Vault Token
 The plugin tries to cache the Vault token obtained from logging into Vault on the `argocd-repo-server`'s container's disk, at `~/.avp/config.json` for the duration of the token's lifetime. This of course requires that the container user is able to write to that path. Some environments, like Openshift, will force a random user for containers to run with; therefore this feature will not work, and the plugin will attempt to login to Vault on every run. This can be fixed by ensuring the `argocd-repo-server`'s container runs with the user `argocd`.
+
+#### Running argocd-vault-plugin in a sidecar container
```

Comment:
Mention the timeouts and how they may need to be adjusted if lots of placeholders: https://argo-cd.readthedocs.io/en/stable/user-guide/config-management-plugins/#using-a-cmp